packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics-latencies.ts

```diff
@@ -270,6 +270,14 @@ export default class CallDiagnosticLatencies extends WebexPlugin {
     return this.precomputedLatencies.get('internal.client.pageJMT') || undefined;
   }
 
+  /**
+   * Download Time JMT
+   * @returns - latency
+   */
+  public getDownloadTimeJMT() {
+    return this.precomputedLatencies.get('internal.download.time') || undefined;
```

Comment:
Is there a reason why this is a `Map` class object? I've noticed this throughout the codebase in areas where a regular object/array would have worked.

Assuming this is potentially an artifact from when this was attempted to be a singleton/instance hybrid plugin?

Since these keys are required to exist as strings, have he considered moving them to `constants`?

Reply:
I don't remember exactly why I decided to use Map, but I don't think was a matter of one better than other. I agree both can work.

Not sure it had to with the singleton issue.

the latencyTimestamps are typed, so no need for constants.
the precomputedLatencies are not typed, but I can create a type if needed.

Reply:
In this case, constants are definitely needed, an engineer needs the exact string `internal.download.time` to receive the **internal download time**. If this was keyed in TS as:

```ts
interface PrecomputedLatencies {
  'internal.download.time': number;
}
```

or even Mapped as:

```ts
export type PrecomputedLatancy = 'internal.download.time';

/* ... somewhere else in code ... */

precomputedLatencies: Map<PrecomputedLatancy, number>;
```

This would allow for at least the keys to be inferred (partially). This also applies to a later comment.

Reply:
added types for the precomputed map